wafv2/rate-based-acl/custom-rule-group.tf

```diff
@@ -0,0 +1,100 @@
+resource "aws_wafv2_rule_group" "rate-based_rule-group" {
+  for_each    = length(var.config) == 0 ? [] : toset(["rate-based-rule-group"])
+  name        = length(var.name_prefix) == 0 ? "rate-based-rule-group" : "${var.name_prefix}-rate-based-rule-group"
+  description = "Rate based custom rule group"
+  scope       = "REGIONAL"
+  capacity    = var.capacity
+
+
+  dynamic "rule" {
+    for_each = { for idx, cfg in var.config : idx => cfg }
+
+    content {
+      name     = rule.value["rule_name"]
+      priority = rule.key
+
+      dynamic "action" {
+        for_each = rule.value["action"] == "allow" ? ["allow"] : []
+        content {
+          allow {}
+        }
+      }
+
+      dynamic "action" {
+        for_each = rule.value["action"] == "block" ? ["block"] : []
+        content {
+          block {}
+        }
+      }
+
+      dynamic "action" {
+        for_each = rule.value["action"] == "count" ? ["count"] : []
+        content {
+          count {}
+        }
+      }
+
+      dynamic "action" {
+        for_each = rule.value["action"] == "captcha" ? ["captcha"] : []
+        content {
+          captcha {}
+        }
+      }
+
+      statement {
+        rate_based_statement {
+
+          limit              = rule.value["limit"]
+          aggregate_key_type = rule.value["aggregate_key_type"]
+
+          dynamic "scope_down_statement" {
+            for_each = length(rule.value["paths"]) > 0 ? ["scope_down_statement"] : []
+            content {
+              or_statement {
```

Comment:
will not work if case of one path:
` OR statement with only one nested statement`